model-optimizer/extensions/front/onnx/quantize_linear_resolver.py

```diff
@@ -54,16 +62,56 @@ def replace_op(self, graph: Graph, node: Node):
+
 
         cast = Cast(graph, {'dst_type': zero_point_type, 'name': node_name + '/Cast'}).create_node()
         rename_nodes([(node, node_name + '/TBD'), (cast, node_name)])
         fake_quantize.out_port(0).connect(cast.in_port(0))
 
+        axis = node.soft_get('axis', 1)
+        if axis is not None and axis != 1:
+            data_shape_node = Shape(graph, {'name': node_name + '/Shape'}).create_node()
+            fake_quantize.in_port(0).get_source().connect(data_shape_node.in_port(0))
+            gather_node = create_op_with_const_inputs(graph, Gather, {1: int64_array([axis]),
+                                                                      2: int64_array(0)},
+                                                      {'name': node_name + '/Gather'})
+            data_shape_node.out_port(0).connect(gather_node.in_port(0))
+
+            rank_node = Rank(graph, {'name': node_name + '/Rank'}).create_node()
+            fake_quantize.in_port(0).get_source().connect(rank_node.in_port(0))
+
+            range_node = create_op_with_const_inputs(graph, Range, {0: int64_array(0),
+                                                                    2: int64_array(1)},
+                                                     {'name': node_name + '/Range'})
+            rank_node.out_port(0).connect(range_node.in_port(1))
+            greater_equal_node = create_op_with_const_inputs(graph, GreaterEqual, {1: int64_array([0])},
+                                                             {'name': node_name + '/GreaterEqual'})
+            range_node.out_port(0).connect(greater_equal_node.in_port(0))
+
+            scatter_elements_node = create_op_with_const_inputs(graph, ScatterElementsUpdate, {1: int64_array([axis]),
+                                                                                               3: int64_array(0)},
+                                                                {'name': node_name + '/ScatterElements'})
+
+            cast_node = Cast(graph, {'dst_type': np.int64, 'name': node_name + '/Cast'}).create_node()
+            greater_equal_node.out_port(0).connect(cast_node.in_port(0))
+            cast_node.out_port(0).connect(scatter_elements_node.in_port(0))
+
+            gather_node.out_port(0).connect(scatter_elements_node.in_port(2))
+            reshape_low_node = Reshape(graph, {'name': node_name + '/Reshape_low'}).create_node()
+            mul_low.out_port(0).connect(reshape_low_node.in_port(0))
+            scatter_elements_node.out_port(0).connect(reshape_low_node.in_port(1))
+            reshape_low_node.out_port(0).connect(fake_quantize.in_port(1))
+
+            reshape_high_node = Reshape(graph, {'name': node_name + '/Reshape_high'}).create_node()
+            mul_high.out_port(0).connect(reshape_high_node.in_port(0))
+            scatter_elements_node.out_port(0).connect(reshape_high_node.in_port(1))
+            reshape_high_node.out_port(0).connect(fake_quantize.in_port(2))
```

Comment:
Code inside this file and model-optimizer/extensions/front/onnx/dequantize_linear_resolver.py is so similar. At least, firtst 27 lines are the exactly the same. Please, refactor the solution to avoid code duplication.